registry/quilt_server/views.py

```diff
@@ -762,6 +761,18 @@ def package_put(owner, package_name, package_hash=None, package_path=None):
             team_access = Access(package=package, user=TEAM)
             db.session.add(team_access)
     else:
+        team_access = (
+                Access.query
+                .filter(sa.and_(
+                    Access.package == package,
+                    Access.user == TEAM
+                ))
+                .one_or_none()
+            )
+        if not team_access and g.auth.user != owner:
```

Comment:
Kind of a corner case, but: if a package has been shared with your team, you can push it. But it's been shared with just you, then you cannot. Should this be a normal access check instead?